src/test/java/com/bakdata/common_kafka_streams/integration/CleanUpTest.java

```diff
@@ -0,0 +1,173 @@
+        this.kafkaCluster.start();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS));
+
+        this.app = new WordCount();
+        this.app.setSchemaRegistryUrl(this.schemaRegistryMockExtension.getUrl());
+        final String inputTopicName = "input";
+        this.app.setInputTopic(inputTopicName);
+        final String outputTopicName = "output";
+        this.app.setOutputTopic(outputTopicName);
+        this.app.setBrokers(this.kafkaCluster.getBrokerList());
+        this.app.setProductive(false);
+        this.app.setStreamsConfig(
+                Map.of("default.value.serde", "org.apache.kafka.common.serialization.Serdes$StringSerde",
+                        "default.key.serde", "org.apache.kafka.common.serialization.Serdes$StringSerde",
+                        StreamsConfig.CACHE_MAX_BYTES_BUFFERING_CONFIG, "0"));
+
+        this.kafkaCluster.createTopic(TopicConfig.forTopic(this.app.getOutputTopic()).useDefaults());
+        this.kafkaCluster.createTopic(TopicConfig.forTopic(this.app.getInputTopic()).useDefaults());
+    }
+
+    @AfterEach
+    void teardown() throws InterruptedException {
+        this.app.close();
+        this.app.getStreams().cleanUp();
+        Thread.sleep(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS));
+        this.kafkaCluster.stop();
+    }
+
+
+    @Test
+    void shouldDeleteState() throws InterruptedException {
+        final SendValuesTransactional<String> sendRequest = SendValuesTransactional
+                .inTransaction(this.app.getInputTopic(), List.of("blub", "bla", "blub"))
+                .useDefaults();
+        this.kafkaCluster.send(sendRequest);
+        Thread.sleep(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS));
+
+        final List<KeyValue<String, Long>> expectedValues = List.of(
+                new KeyValue<>("blub", 1L),
+                new KeyValue<>("bla", 1L),
+                new KeyValue<>("blub", 2L)
+        );
+
+        this.runAndAssertContent(expectedValues, "First run");
+        Thread.sleep(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS));
+        this.runCleanUp();
+
+        final List<KeyValue<String, Long>> collect = expectedValues.stream()
+                .flatMap(entry -> Stream.of(entry, entry))
+                .collect(Collectors.toList());
+        this.runAndAssertContent(collect, "Run after clean up");
+
+    }
+
+    @Test
+    void shouldReprocessAlreadySeenRecords() throws InterruptedException {
+        final SendValuesTransactional<String> sendRequest =
+                SendValuesTransactional.inTransaction(this.app.getInputTopic(),
+                        Arrays.asList("a", "b", "c")).useDefaults();
+        this.kafkaCluster.send(sendRequest);
+
+        this.runAndAssertSize(3);
+        this.runAndAssertSize(3);
+
+        // Wait until all stream application are completely stopped before triggering cleanup
+        Thread.sleep(TimeUnit.SECONDS.toMillis(TIMEOUT_SECONDS));
+        this.runCleanUp();
+        this.runAndAssertSize(6);
```

Comment:
Why should there be 6 records after cleaning up and not sending any records?

Reply:
The cleanUp doesn't affect the output topic.  It contains already 3 elements from the run before. After the cleanUp, the application reprocesses all elements from the input topic and writes them to the output topic.

Reply:
So the input topic is not cleaned?

Reply:
The input topic's content is not deleted, but the offset is reset.

Reply:
Ok but we need to clear either the input or output topic

Reply:
Clearing the output topic would be perfect. In case we change something internally we would consume the input from the beginning and only the new data is in the output topic